odl/tomo/util/source_detector_shifts.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2014-2020 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Source and detector shifts for divergent beam geometries."""
+
+from __future__ import print_function, division, absolute_import
+import numpy as np
+from odl.discr import uniform_discr_frompartition
+from odl.discr.discr_utils import nearest_interpolator
+
+__all__ = ('flying_focal_spot')
+
+
+def flying_focal_spot(angle, apart, shifts):
+    """Flying focal spot shifts for divergent beam geometries.
+
+    Parameters
+    ----------
+    angle : float or `array-like`
+        Angle(s) in radians describing the counter-clockwise
+        rotation of source and detector.
+    apart : 1-dim. `RectPartition`
+        Partition of the angle interval.
+    shifts : sequence of `array-like`
+        Each vectors in a sequence represent a subsequent shift
+        relative to the default source position. Vector elements
+        represent shifts along the following directions:
+        det_to_src, tangent to the rotation, rotation axis.
+    """
+    angle = np.array(angle, dtype=float, copy=False, ndmin=1)
+    interpolator = nearest_interpolator(np.arange(apart.size),
```

Comment:
We do support multiple dimensions in angle partitions, which is broken here. Document and assert correctness.